### PR TITLE
Update parsing and calculation of metrics from bcl2fastq

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR updates the population of the SampleLaneSequencingMetrics model to only include the required fields and removes logic that is now unnecessary for parsing bcl2fastq data and calculating metrics.

### Fixed
- Population of `SampleLaneSequencingMetrics` from bcl2fastq data.

### CODE REVIEW

File: cg/apps/sequencing_metrics_parser/parsers/bcl2fastq_to_sequencing_statistics.py [hunk 1]
```diff
@@ -35,23 +31,21 @@ def get_sequencing_metrics_from_bcl2fastq(stats_json_path: str):
...
 
     for conversion_result in raw_sequencing_metrics.conversion_results:
         for demux_result in conversion_result.demux_results:
-            statistics_for_sample_in_lane: SampleLaneSequencingMetrics = (
-                create_sequencing_statistics(
-                    conversion_result=conversion_result,
-                    demux_result=demux_result,
-                    raw_sequencing_metrics=raw_sequencing_metrics,
-                )
+            metrics_for_sample_in_lane: SampleLaneSequencingMetrics = create_sequencing_metrics(
```
Comment (ChrOertlin) [inline cg/apps/sequencing_metrics_parser/parsers/bcl2fastq_to_sequencing_statistics.py:38]:
```suggestion
            metrics_for_sample_in_lane: SampleLaneSequencingMetrics = create_sample_lane_sequencing_metrics(
```

File: cg/apps/sequencing_metrics_parser/parsers/bcl2fastq_to_sequencing_statistics.py [hunk 2]
```diff
@@ -74,57 +68,21 @@ def create_sequencing_statistics(
         SampleLaneSequencingMetrics: A SampleLaneSequencingMetrics object that encapsulates the statistics for a sample
         in a lane on the flow cell.
     """
-    yield_in_megabases: float = calculate_yield_in_megabases(conversion_result=conversion_result)
-
-    passed_filter_percent: float = calculate_pass_filter_ratio(conversion_result=conversion_result)
 
     average_quality_score: float = calculate_average_quality_score(demux_result=demux_result)
-
     bases_with_q30_percent: float = calculate_q30_ratio(demux_result=demux_result)
```
Comment (ChrOertlin) [inline cg/apps/sequencing_metrics_parser/parsers/bcl2fastq_to_sequencing_statistics.py:73]:
```suggestion
    bases_with_q30_percent: float = calculate_per_lane_q30_ratio(demux_result=demux_result)
```

File: cg/apps/sequencing_metrics_parser/parsers/bcl2fastq_to_sequencing_statistics.py [hunk 3]
```diff
@@ -74,57 +68,21 @@ def create_sequencing_statistics(
         SampleLaneSequencingMetrics: A SampleLaneSequencingMetrics object that encapsulates the statistics for a sample
         in a lane on the flow cell.
     """
-    yield_in_megabases: float = calculate_yield_in_megabases(conversion_result=conversion_result)
-
-    passed_filter_percent: float = calculate_pass_filter_ratio(conversion_result=conversion_result)
 
     average_quality_score: float = calculate_average_quality_score(demux_result=demux_result)
```
Comment (ChrOertlin) [inline cg/apps/sequencing_metrics_parser/parsers/bcl2fastq_to_sequencing_statistics.py:72]:
```suggestion
    average_quality_score: float = calculate_per_lane_average_quality_score(demux_result=demux_result)
```

File: cg/apps/sequencing_metrics_parser/parsers/bcl2fastq_to_sequencing_statistics.py [hunk 4]
```diff
@@ -74,57 +68,21 @@ def create_sequencing_statistics(
...
-        [metric.mismatch_counts.get("0", 0) for metric in demux_result.index_metrics]
-    )
-    total_reads: int = demux_result.number_reads
-
-    return perfect_reads_ratio(perfect_reads=perfect_reads, total_reads=total_reads)
-
-
 def calculate_q30_ratio(demux_result: DemuxResult) -> float:
     """
     Calculate the proportion of bases that have a Phred quality score of 30 or more (Q30) for a sample
```
Comment (ChrOertlin) [inline cg/apps/sequencing_metrics_parser/parsers/bcl2fastq_to_sequencing_statistics.py:88]:
```suggestion
    Calculate the proportion of bases that have a Phred quality score of 30 or more (Q30) for a sample in a flow cell lane
```

right?

**Review by ChrOertlin — APPROVED** [review]
Loving all the red. :P 

Just a comment on the calculation functions, should we specificy them to be per lane calulcations?

**Comment by seallard** [comment]
Deployed to prod.

**Pull request merged**